git/git.go

```diff
@@ -28,7 +28,7 @@ func Dir() (string, error) {
 		return cachedDir, nil
 	}
 
-	dirCmd := gitCmd("rev-parse", "-q", "--git-dir")
+	dirCmd := gitCmd("rev-parse", "-q", "--git-common-dir")
```

Comment:
What is the earliest git version where `--git-common-dir` was introduced?

Reply:
Looks like it was first introduced in https://git-scm.com/docs/git-rev-parse/2.5.1.

Reply:
Based on these two pages, I'm inclined to say that it will always work, even if the path is relative.

https://git-scm.com/docs/git

`GIT_COMMON_DIR`
> If this variable is set to a path, non-worktree files that are normally in $GIT_DIR will be taken from this path instead. Worktree-specific files such as HEAD or index are taken from $GIT_DIR

`commondir`
>If this file exists, $GIT_COMMON_DIR (see git[1]) will be set to the path specified in this file if it is not explicitly set. If the specified path is relative, it is relative to $GIT_DIR. The repository with commondir is incomplete without the repository pointed by "commondir"

Reply:
If you're setting `GIT_COMMON_DIR`, then based on the docs, it seems like you should be expecting this dir to be used for all non-worktree files. I guess your question though is concerned with what other code is using `Dir()`? So, what other hub features rely on `GIT_DIR`, and do they ever expect to use worktree-specific files?

Reply:
Requiring git 2.5 all of the sudden is a dealbreaker, since thus far we've supported git 1.7+

Reply:
Are you against something like...
```
        dirCmd := gitCmd("rev-parse", "-q", "--git-dir")
	parseCmd.Stderr = nil
	gitDir, err := parseCmd.Output()

	parseCmd := gitCmd("rev-parse", "--is-inside-worktree")
	parseCmd.Stderr = nil
	isWorktree, err := parseCmd.Output()

        if (isWorkTree) {
            // assume worktree and use GIT_DIR/../../
            gitDir = gitDir + "/../../"

            // check if path exists, otherwise fallback to regular behavior
        }
```

Reply:
Okay, so git worktrees aren't support in git@v.1.7.0. Any hub command attempted from a worktree in git@1.7.0 will fail.

So, changing this code to use ```git rev-parse --git-common-dir``` instead of ```git rev-parse --git-dir``` will not change the behavior in any git release that does not include worktree support. 

![image](https://user-images.githubusercontent.com/22828293/75586880-d49ec780-5a3a-11ea-87f6-0be6ade62e0c.png)

![image](https://user-images.githubusercontent.com/22828293/75586886-d9637b80-5a3a-11ea-8fa3-6bd699dd7c54.png)

Reply:
Okay so now looking at the first release of git with worktrees.

![image](https://user-images.githubusercontent.com/22828293/75587833-fdc05780-5a3c-11ea-85ff-2cafa9dce4c5.png)

Based on these results, I would say there is no issue in changing ```git rev-parse --git-dir``` to ```git rev-parse --git-common-dir```. If you are in a worktree prior to git@v2.6.0, you ```rev-parse``` command will fail anyway.

Reply:
> Based on these results, I would say there is no issue in changing `git rev-parse --git-dir` to `git rev-parse --git-common-dir`. If you are in a worktree prior to git@v2.6.0, your `rev-parse` command will fail anyway.

That's a good point; thank you for looking into this in such depth!

I'm wondering what about when you're _not_ in a worktree, but in a regular git working directory, and you issue `git rev-parse --git-common-dir` with an old git version. Wouldn't it fail because `--git-common-dir` flag is not recognized there?

Reply:
lol yes you're right...I got so hung up on worktrees that I forgot about the nomal scenario.